src/main/kotlin/org/jetbrains/research/ml/kotlinAnalysis/Pipeline.kt

```diff
@@ -0,0 +1,37 @@
+package org.jetbrains.research.ml.kotlinAnalysis
+
+import com.intellij.ide.impl.ProjectUtil
+import java.io.File
+import java.nio.file.Files
+import java.nio.file.Path
+
+class Pipeline(private val outputDir: Path) {
+    private val indexer = IndexBuilder()
+    private val psiProvider = PsiProvider()
+    private val adapter = CloneDetectionAdapter()
+
+    fun extractMethodsToCloneDetectionFormat(inputDir: Path) {
+        val projectIndexWriter = File(outputDir.toFile(), "project_index.txt").printWriter()
+        val methodIndexWriter = File(outputDir.toFile(), "method_index.csv").printWriter()
+        val methodDataWriter = File(outputDir.toFile(), "method_data.txt").printWriter()
```

Comment:
Maybe is it better to create arguments with the names of these files?

Reply:
Probably it would be better create a function to extract PrintWriter from the file path and use it here?

Reply:
Probably it would be great to add three new class fields for these writers